### DISCUSSION TITLE
Support for Stripe Checkout

### DISCUSSION THREAD

**Discussion by DotCat1985** [OrchardCMS/OrchardCore.Commerce#400] [Q&A] [answered]
Does the module **OrchardCore.Commerce.Payment.Stripe** support [Stripe Checkout](https://stripe.com/docs/payments/checkout)?

**Comment by sarahelsaig** [comment, marked as answer]
No, only [Stripe Payment Element](https://stripe.com/docs/payments/payment-element) was implemented.

**Reply by DotCat1985** [reply]
Does [Stripe Payment Element](https://stripe.com/docs/payments/payment-element) accept payments with PayPal?

**Reply by sarahelsaig** [reply]
I think it's possible ([see here](https://stripe.com/docs/payments/payment-element#payment-methods)), but I haven't tested it because it requires [Stripe Connect](https://stripe.com/connect) which I didn't need in the past so it's not set up on my account. Check out Stripe's documentation [here](https://stripe.com/docs/payments/paypal/accept-a-payment) about the PayPal payment provider. You can configure it [on their dashboard](https://dashboard.stripe.com/), so if you need help please contact their support.

The payment element in OCC has dynamic payment method configuration. This means once you've connected your Stripe account to your OCC site ([see our documentation](https://commerce.orchardcore.net/en/latest/features/stripe-payment/)) all other configuration is done outside OCC, on the Stripe dashboard.

**Reply by DotCat1985** [reply]
Does the module **OrchardCore.Commerce.Payment.Stripe** support the using of promotion codes by using **Stripe Elements**?

**Reply by sarahelsaig** [reply]
No, the promo code feature in Stripe Elements is still in beta. Also it's not a good fit for OCC's flow so I don't think we will implement it in the future. In OCC the promotions are part of the price calculation and separate from payment. OrchardCore.Commerce.Payment.Stripe will never need to know about what kind of promotions are used because it receives the final prices where promotions and taxes are already applied.

Promo codes may be implemented by a new `IPromotionProvider` service. (Maybe something similar to the [Global Discounts](https://commerce.orchardcore.net/en/latest/features/promotions/#global-discounts) feature.) Feel free to open an issue with details about what you'd like from such a provider.
